### ISSUE TITLE
sanity tests - cache and paging

### ISSUE THREAD

**Issue by dlangille** [FreshPorts/freshports#451]
There is no paging on https://www.freshports.org/sanity_test_failures.php

The cache does not seem to be automatically cleared:

* commit delete
* new sanity test failure

**Comment by dlangille** [comment]
This was fixed by amending the existing `CommitsCacheClear()` function in `fp-listen.py`

Every time a commit is amended/deleted, this function will clear the following from cache:

existing:

```
COMMIT_CACHE_PATH      = %(BASEDIR)s/cache/commits/%%s/
```

new:

```
SANITY_CACHE_PATH      = %(BASEDIR)s/cache/general/sanity_test_failures.%%s.*.html
SANITY_MAIN_CACHE_PATH = %(BASEDIR)s/cache/general/sanity_test_failures.PageSize*.html
```

Where `%%s` is the commit id.

**Comment by dlangille** [comment]
This does mean that the sanity test list is cleared each time a commit occurs. That is overkill.  However, going the other way creates more work and more code. Simple cache clearing is better cache clearing.

**Comment by dlangille** [comment]
Fixed with release 1.0.7 - that code is not on github.

**Comment by dlangille** [comment]
Installed in prod by jesse.
